Common subdirectories: samples/webview/.vs and samples/webview - bug/.vs
Common subdirectories: samples/webview/Backup and samples/webview - bug/Backup
Only in samples/webview - bug/: vc_mswu_x64
Common subdirectories: samples/webview/vc_mswud_x64 and samples/webview - bug/vc_mswud_x64
diff -u samples/webview/webview.cpp "samples/webview - bug/webview.cpp"
--- samples/webview/webview.cpp	2021-10-15 13:48:56.454439900 +1100
+++ "samples/webview - bug/webview.cpp"	2021-10-15 13:18:41.574649400 +1100
@@ -41,6 +41,7 @@
 #include "wx/fs_arc.h"
 #include "wx/fs_mem.h"
 #include "wx/stdpaths.h"
+#include "wx/listctrl.h""
 
 #ifndef wxHAS_IMAGES_IN_RESOURCES
     #include "../sample.xpm"
@@ -243,6 +244,7 @@
     wxStaticText* m_info_text;
     wxTextCtrl* m_find_ctrl;
     wxToolBar* m_find_toolbar;
+	wxListView* m_list_view;
 
     wxMenuHistoryMap m_histMenuItems;
     wxString m_findText;
@@ -353,7 +355,6 @@
     // Create find control.
     m_find_ctrl = new wxTextCtrl(m_find_toolbar, wxID_ANY, wxEmptyString, wxDefaultPosition, wxSize(140,-1), wxTE_PROCESS_ENTER);
 
-
     //Find options menu
     wxMenu* findmenu = new wxMenu;
     m_find_toolbar_wrap = findmenu->AppendCheckItem(wxID_ANY,"Wrap");
@@ -373,6 +374,30 @@
     m_find_toolbar_options->SetDropdownMenu(findmenu);
     m_find_toolbar->Realize();
 
+    // Create panel for listview
+    wxPanel* list_panel = new wxPanel(this);
+    topsizer->Add(list_panel, wxSizerFlags().Expand());
+    // Create sizer for panel.
+    wxBoxSizer* list_panel_sizer = new wxBoxSizer(wxVERTICAL);
+    list_panel->SetSizer(list_panel_sizer);
+#define LIST_ID 9999
+	m_list_view = new wxListView(list_panel, LIST_ID);
+	m_list_view->SetSingleStyle(wxLC_SINGLE_SEL);
+    list_panel_sizer->Add(m_list_view, wxSizerFlags().Expand());
+	m_list_view->InsertColumn(0, "");
+	m_list_view->InsertItem(0, "Item 0");
+	m_list_view->InsertItem(1, "Item 1");
+	m_list_view->InsertItem(2, "Item 2");
+	m_list_view->InsertItem(3, "Item 3");
+	m_list_view->InsertItem(4, "Item 4");
+	Bind(wxEVT_LIST_ITEM_SELECTED, [this](wxListEvent& e) {
+		wxLogMessage("Got list item selected event index=" + wxString::Format(wxT("%d"), e.GetIndex()));
+		RunScript("function f(a){for (i=0; i<10000000; ++i) {x=a+i;} return x;}f('Hello World!');");
+	}, LIST_ID);
+	Bind(wxEVT_LIST_BEGIN_DRAG, [this](wxListEvent& e) {
+		wxLogMessage("Got drag event for index=" + wxString::Format(wxT("%d"), e.GetIndex()));
+	}, LIST_ID);
+
     // Create the info panel
     m_info = new wxInfoBar(this);
     topsizer->Add(m_info, wxSizerFlags().Expand());
@@ -393,7 +418,7 @@
     }
 #endif
     // Create the webview
-    m_browser = wxWebView::New();
+    m_browser = wxWebView::New(wxWebViewBackendEdge);
 #ifdef __WXMAC__
     // With WKWebView handlers need to be registered before creation
     m_browser->RegisterHandler(wxSharedPtr<wxWebViewHandler>(new wxWebViewArchiveHandler("wxfs")));
@@ -1119,7 +1144,7 @@
 
 void WebFrame::OnRunScriptString(wxCommandEvent& WXUNUSED(evt))
 {
-    RunScript("function f(a){return a;}f('Hello World!');");
+    RunScript("function f(a){for (i=0; i<100000000; ++i) {x=a+i;} return x;}f('Hello World!');");
 }
 
 void WebFrame::OnRunScriptInteger(wxCommandEvent& WXUNUSED(evt))
